app/shared/sentry.ts

```diff
@@ -0,0 +1,59 @@
+import * as Sentry from "@sentry/browser";
+import { Extras } from "@sentry/types";
+import { BrowserTracing } from "@sentry/tracing";
+import env from "~/shared/env";
+
+const SENTRY_DSN =
+  typeof window !== "undefined" ? env.SENTRY_DSN : process.env.SENTRY_DSN;
+const VERCEL_ENV =
+  typeof window !== "undefined" ? env.VERCEL_ENV : process.env.VERCEL_ENV;
+export const initSentry = () => {
+  SENTRY_DSN
+    ? Sentry.init({
+        dsn: SENTRY_DSN,
+        integrations: [new BrowserTracing()],
+        tracesSampleRate: 1.0,
+        environment: VERCEL_ENV || "development",
+      })
+    : () => null;
+};
```

Comment:
1 - Without the window check I got a window is not defined so we need both  :/
2- VERCEL_ENV is a reserved keyword env by vercel that allows us to know if we are in preview or production

https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables

Reply:
here it is supposed to just work, with no window checks

Reply:
Moved to the getter